asserts/model_test.go

```diff
@@ -840,6 +843,21 @@ func (mods *modelSuite) TestCore20GradeDangerous(c *C) {
 	})
 }
 
+func (mods *modelSuite) TestCore20ValidStorageSafety(c *C) {
+	encoded := strings.Replace(core20ModelExample, "TSLINE", mods.tsLine, 1)
+	encoded = strings.Replace(encoded, "OTHER", "", 1)
+	encoded = strings.Replace(encoded, "grade: secured\n", "grade: signed\n", 1)
+
+	for _, sss := range []string{"prefer-encrypted", "prefer-unencrypted", "encrypted"} {
+		ex := strings.Replace(encoded, "storage-safety: encrypted\n", fmt.Sprintf("storage-safety: %s\n", sss), 1)
+		a, err := asserts.Decode([]byte(ex))
+		c.Assert(err, IsNil)
+		c.Check(a.Type(), Equals, asserts.ModelType)
+		model := a.(*asserts.Model)
+		c.Check(string(model.StorageSafety()), Equals, sss)
```

Comment:
I guess I would prefer that we check strictly against the value of the variables model.StorageSafetyEncrypted, etc. than the string, because we could have a bug verifying the values allowed for the header, such that something like `preffffer-unencrypted` is allowed, and if we make the typo to `preffffer-unencrypted` in the declaration of this loop, then this test would still pass, where if we have that same bug, but check against the variables from the package directly, the test would now fail (unless we also made the same _exact_ typo in the definition of the variables themselves which seems unlikely).

hopefully that stream of consciousness makes sense :-)

Reply:
I agree

Reply:
Not sure I 100% understood but I updatd the code with what I think is suggested here. Please correct me if I'm wrong.